src/neo/SmartContract/Helper.cs

```diff
@@ -164,7 +166,8 @@ internal static bool VerifyWitnesses(this IVerifiable verifiable, StoreView snap
                 }
                 using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, verifiable, snapshot.Clone(), gas))
                 {
-                    engine.LoadScript(verification, CallFlags.None).InstructionPointer = offset;
+                    engine.LoadScript(verification, CallFlags.ReadOnly).InstructionPointer = offset;
```

Comment:
If you change it to `ReadOnly`, we have to reverify every transactions for each block. This will significantly reduce tps.

Reply:
But checkwitness require states...

Reply:
Then don't call checkwitness in `verify()`.

Reply:
But we need to be able to check if it was signed by the owner. Maybe we can have two checkwitness one without storage access, or check the flag only when it's needed, like in this PR